torch/distributed/fsdp/flat_param.py

```diff
@@ -2385,6 +2402,13 @@ def _construct_padding_tensor(
     )
 
 
+# Use `lru_cache(1)` to only log the warning once (assuming the fixed warning
```

Comment:
nit: can't we use warnings API to warn once?

Reply:
I am actually not sure how the logger and the `warnings` module are different. I see internally they use the logger for everything, so I decided to start using the logger for FSDP.